homeassistant/components/rachio/switch.py

```diff
@@ -509,3 +544,96 @@ async def async_added_to_hass(self) -> None:
                 self.hass, SIGNAL_RACHIO_SCHEDULE_UPDATE, self._async_handle_update
             )
         )
+
+
+class RachioValve(SwitchEntity):
+    """Representation of one smart hose timer valve."""
+
+    _attr_icon = "mdi:water"
+
+    def __init__(self, person, base, data) -> None:
+        """Initialize a new smart hose valve."""
+        self._person = person
+        self._base = base
+        self.id = data[KEY_ID]
+        self._attr_name = data[KEY_NAME]
+        self._attr_unique_id = f"{self.id}-valve"
+        self._static_attrs = data[KEY_STATE][KEY_REPORTED_STATE]
+        self._detect_flow = data[KEY_DETECT_FLOW]
+        self._attr_is_on = self._static_attrs.get(KEY_CURRENT_STATUS) is not None
+        self._battery_status = self._static_attrs[KEY_BATTERY_STATUS]
+        self._running_attrs = self._static_attrs.get(KEY_CURRENT_STATUS, {})
+        self._attr_should_poll = True
+        self._just_toggled = False
+        self.skip_throttle = False
+
+        self._attr_device_info = DeviceInfo(
+            identifiers={
+                (
+                    DOMAIN_RACHIO,
+                    self._base.serial_number,
+                )
+            },
+            connections={(CONNECTION_NETWORK_MAC, self._base.mac_address)},
+            manufacturer=DEFAULT_NAME,
+            model="Smart Hose Timer",
+            name="Smart Hose Timer Base Station",
+        )
+
+    @property
+    def extra_state_attributes(self) -> dict[str, Any]:
+        """Return the optional state attributes."""
+        _runtime = int(self._running_attrs.get(KEY_DURATION_SECONDS, 0))
+        props = {
+            "Flow Detection Enabled": self._detect_flow,
+            "Battery Status": self._battery_status,
+        }
+        if self._attr_is_on:
+            if _runtime > 0:
+                props[ATTR_DURATION] = f"{round(_runtime / 60)} minutes"
+            props[ATTR_WATERING_REASON] = self._running_attrs.get(KEY_REASON)
+            props[ATTR_FLOW_DETECTED] = self._running_attrs.get(KEY_FLOW_DETECTED)
+            props[ATTR_STARTED_WATERING] = self._running_attrs.get(KEY_START_TIME)
+
+        return props
```

Comment:
https://developers.home-assistant.io/docs/core/entity?_highlight=entity#generic-properties

We generally don't add new extra state attributes anymore and create them as separate sensors on the same device

> Entities that generate a significant amount of state changes can quickly increase the size of the database when the extra_state_attributes also change frequently. Minimize the number of extra_state_attributes for these entities by removing non-critical attributes or creating additional sensor entities.

Reply:
Before you get too carried away I totally screwed up my commit history and I need to push another one to get back to where it was.

Reply:
Ok, so all of these should go?

Reply:
I'd drop them from this PR and add them as separate sensors in a followup PR if they are needed